### DISCUSSION TITLE
Polymorphic button raises the warning about deprecated onClick usage

### DISCUSSION THREAD

**Discussion by TwoWheelDev** [heroui-inc/heroui#4360] [Q&A]
I'm fairly new to nextui, and using the the examples from the website for a navbar with a button raises the warning regarding onClick being deprecated. The example shown here: https://nextui.org/docs/components/link#polymorphic-component will raise this warning.

I've read through issue #4292 that is linked in the warning but it doesn't provide any clear indication how to fix this usage (which seems to be rooted within nextui as I don't directly call onClick myself.

How would I go about fixing this?

**Comment by wingkwong** [comment]
which version are you using?

**Reply by TwoWheelDev** [reply]
v2.6.8 (Which seems to be the latest version)

**Reply by wingkwong** [reply]
If it's just the example code, I couldn't reproduce the issue. Can you check if you have other components? If possible, please provide a sandbox for us to investigate. Thanks.

```
    <Button
      showAnchorIcon
      as={Link}
      color="primary"
      href="https://github.com/nextui-org/nextui"
      variant="solid"
    >
      Button Link
    </Button>
```

**Reply by TwoWheelDev** [reply]
So it's using the code from the NavBar example with a button. Which is defined as follows:
```
<NavbarContent justify="end"
        <NavbarItem className="hidden lg:flex">
                  <Link href="#">Login</Link>
        </NavbarItem>
        <NavbarItem>
             <Button as={Link} color="primary" href="#" variant="flat">Sign Up</Button>
         </NavbarItem>
</NavbarContent>
```

Removing `as={Link}` from the above code clears the issue (but as that button will be a link, it's needed!)

**Reply by wingkwong** [reply]
I can reproduce now. The `onClick` is passed from the `Button` component internally.

**Reply by TwoWheelDev** [reply]
Ahh! Good to know then that it wasn't something I'd done!
